version: '3.3'

services:
  plex:
    image: lscr.io/linuxserver/plex:<VERSION/LATEST>
    container_name: plex
    restart: unless-stopped
    ports:
      # CHECK PLEX DOCS, NOT ALL FOLLOWING PORTS NEED TO BE OPEN
      - 32400:32400
      - 1900:1900/udp
      - 3005:3005
      - 5353:5353/udp
      - 8324:8324
      - 32410:32410/udp
      - 32412:32412/udp
      - 32413:32413/udp
      - 32414:32414/udp
      - 32469:32469
    networks:
      - <NETWORK>
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /path/config:/config
      - /path/movies:/movies
      - /path/shows:/shows
      - /path/music:/music
      - /path/tvseries:/tvseries
    environment:
      - PUID=<run id command to get PUID and PGID>
      - PGID=<run id command to get PUID and PGID>
      # - TZ=Europe/Lisbon #optional
      - VERSION=docker
      - PLEX_CLAIM=<PASTE HERE THE CLAIM CODE FROM THE URL BELOW>
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.plex.entrypoints=http"
      - "traefik.http.routers.plex.rule=Host(`URL`)"
      - "traefik.http.middlewares.plex-https-redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.plex.middlewares=plex-https-redirect"
      - "traefik.http.routers.plex-secure.entrypoints=https"
      - "traefik.http.routers.plex-secure.rule=Host(`URL`)"
      - "traefik.http.routers.plex-secure.tls=true"
      - "traefik.http.routers.plex-secure.service=plex"
      - "traefik.http.services.plex.loadbalancer.server.port=32400"
      - "traefik.docker.network=NETWORK"

networks:
  NETWORK:
    external: true

# https://www.plex.tv/claim/